internal/dag/gatewayapi_processor.go

```diff
@@ -205,16 +211,29 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRo
 
 	for _, rule := range route.Spec.Rules {
 
-		var pathPrefixes []string
+		matchconditions := []*matchConditions{}
 		var services []*Service
 
 		for _, match := range rule.Matches {
+			mc := &matchConditions{}
 			switch match.Path.Type {
```

Comment:
probably need a `nil` check on the `Path` field as well, since it can be omitted

probably should add a test that adds a header match but no path match (which should result in a default `/` prefix match) to ensure this is covered

Reply:
Added `match.Path.Value == "" || match.Path.Type == ""` check since `match.Path` was changed to pointer recently by https://github.com/kubernetes-sigs/gateway-api/commit/9d63656df8cc9e67da60d4fe3f6289aad22e72d3 (not included in current gateway-api release) so we need to change it to nilt check when gateway-api lib is updated.